client/src/test/java/org/spine3/change/StringMismatchShould.java

```diff
@@ -0,0 +1,95 @@
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package org.spine3.change;
+
+import com.google.protobuf.StringValue;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.spine3.change.StringMismatch.expectedEmpty;
+import static org.spine3.change.StringMismatch.expectedNotEmpty;
+import static org.spine3.change.StringMismatch.unexpectedValue;
+import static org.spine3.change.StringMismatch.unpackActual;
+import static org.spine3.change.StringMismatch.unpackExpected;
+import static org.spine3.change.StringMismatch.unpackNewValue;
+import static org.spine3.protobuf.AnyPacker.unpack;
+import static org.spine3.test.Tests.hasPrivateUtilityConstructor;
+
+public class StringMismatchShould {
+
+    private static final String EXPECTED = "expected string";
+    private static final String ACTUAL = "actual string";
+    private static final String NEW_VALUE = "new value to set";
+
+    private static final int VERSION = 1;
+
+    @Test
+    public void have_private_constructor() {
+        assertTrue(hasPrivateUtilityConstructor(StringMismatch.class));
+    }
+
+    @Test
+    public void create_instance_for_expected_empty_string() {
+        final ValueMismatch mismatch = expectedEmpty(ACTUAL, NEW_VALUE, VERSION);
+
+        assertEquals("", unpackExpected(mismatch));
+        assertEquals(ACTUAL, unpackActual(mismatch));
+        assertEquals(NEW_VALUE, unpackNewValue(mismatch));
+        assertEquals(VERSION, mismatch.getVersion());
+    }
+
+    @Test
+    public void create_instance_for_unexpected_empty_string() {
+        final ValueMismatch mismatch = expectedNotEmpty(EXPECTED, VERSION);
+
+        assertEquals(EXPECTED, unpackExpected(mismatch));
+
+        // Check that the actual value from the mismatch is an empty string.
+        assertEquals("", unpackActual(mismatch));
+
+        // We wanted to clear the field. Check that `newValue` is an empty string.
+        assertEquals("", unpackNewValue(mismatch));
+
+        // Check version.
+        assertEquals(VERSION, mismatch.getVersion());
+    }
+
+    @Test
+    public void create_instance_for_unexpected_value() {
+        final ValueMismatch mismatch = unexpectedValue(EXPECTED, ACTUAL, NEW_VALUE, VERSION);
+
+        final StringValue expected = unpack(mismatch.getExpected());
+        final StringValue actual = unpack(mismatch.getActual());
+        final StringValue newValue = unpack(mismatch.getNewValue());
+
+        assertEquals(EXPECTED, expected.getValue());
+        assertEquals(ACTUAL, actual.getValue());
+        assertEquals(NEW_VALUE, newValue.getValue());
+        assertEquals(VERSION, mismatch.getVersion());
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void do_not_accept_same_expected_and_actual() {
```

Comment:
Should be `not_accept...`, as the sentence will read `StringMismatch should ... not accept same expected and actual`.